source/jobs/JobEngine.cpp

```diff
@@ -153,6 +153,22 @@ void JobEngine::exec_action(PlainJobDocument::JobAction action, const std::strin
             return;
         }
     }
+    else if (action.type == "runShellCommand")
```

Comment:
why are we having separate block for building command for `shell-command-handler`?  I think without changing anything we can still execute the same handler file. And anyways the handler file name is passed in job document.

Reply:
this is needed because some customers e.g. the customer who reported the ticket, he puts `/bin` as the path, this will cause device client not being able to find `shell-command-handler.sh` under `sample-job-handler` directory

Reply:
Hey @marcoemorais-aws , what are your thought on my this and other comment below? I think if user is passing a different directory path, we should execute the handler present in that directory.

Reply:
It does seem odd that we do not honor the input path for this runShellCommand type. I think we need to look at this again and make a decision for whether we really need this unique call to buildCommand

Reply:
One thing we discussed before was just having a handler that just builds the command and invokes the next handler... We could just pass all runHandler jobs to this preHandler and it check user and format the command and then invoke the handler in the path.